### PR TITLE
Added upper limit to browse by date (year)

### PR DESCRIPTION
**Pull request by mspalti**
## Description
Attempts to set an upper limit to the browse by year options. So for example if a collection contains publications from 2001- 2010 the first year in the dropdown list is 2010 (not 2023). 

List of changes in this PR:
* Modified the browse-by-date-page.component to set the upper limit by checking the date field of the most recent item.
* Modified the browse.service function to accept an optional SortDirection parameter.

To test this, browse a collection in which no item has a date.issued value from 2023. The first year in the pulldown should match the most recent item in the collection. Note that if the most recent item doesn't have a properly formatted date then the year option will default to the current (2023) year.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [ESLint](https://eslint.org/) validation using `yarn lint`
- [x] My PR doesn't introduce circular dependencies (verified via `yarn check-circ-deps`)
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [ ] If my PR includes new libraries/dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [ ] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [ ] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Comment by mspalti** [comment]
@tdonohue, fyi, the test failure for this PR appeared after I rebased main.  I also see the same error intermittently when I build and test my copy of the main branch.

Besides the test, the metadata browse component UI isn't working  for my main branch. The "back to results" button updates the page with items (not metadata). That actually aligns with the test failure.  But I don't see the same UI result on the demo site, which is odd.

I think problem is related to #2139, which was recently merged. Given how the browse-by-metadata component works it's necessary to always set the `value` property in the subscription rather than reset it conditionally when `typeof params.value === 'string'`.  That approach means it isn't reset when the `params.value` changes to `undefined`.  

If that's the case then I suppose the tests might fail intermittently due to execution order.

I went ahead and reverted the one line change in `browse-by-metadata-page.component.ts`. Noting that here in case it needs discussion. Looks like the change wasn't essential for that PR since it's focused on the `startsWith` parameter, but I could be missing something.

**Comment by mspalti** [comment]
@artlowel , I just verified a bug in main that was introduced in #2139. It popped up as a random test error when I was working on this PR. The problem is that the "all browse results" back button no longer works because of this change line: https://github.com/DSpace/dspace-angular/blob/1ddeeed081034bf2f2a3f527309bf41d3827cc5a/src/app/browse-by/browse-by-metadata-page/browse-by-metadata-page.component.ts#L155
It's inside a subscription and no longer updates when `params.value` emits as undefined. I think that's causing the back button to return to a list of items instead of metadata values. Do you agree? If so, I've gone ahead and reverted that change in this PR.

File: src/app/browse-by/browse-by-metadata-page/browse-by-metadata-page.component.ts [hunk 1]
```diff
@@ -152,9 +152,7 @@ export class BrowseByMetadataPageComponent implements OnInit, OnDestroy {
           this.browseId = params.id || this.defaultBrowseId;
           this.authority = params.authority;
 
-          if (typeof params.value === 'string'){
-            this.value = params.value.trim();
-          }
+          this.value = +params.value || params.value || '';
```
Comment (tdonohue) [inline src/app/browse-by/browse-by-metadata-page/browse-by-metadata-page.component.ts:155]: I think reverting this line is still problematic, as the `+params.value` will remove leading zeros.  If you search for "0905" it trims it to "905".  So, we may need to fix this in a different way.

**Review by tdonohue — COMMENTED** [review]
@mspalti : Tested this today and it works as described.  I think you are correct that you've found a bug in `browse-by-metadata-page` updates... but, I'm not sure reverting it is the correct fix. See comment above and my inline comment below.

File: src/app/browse-by/browse-by-metadata-page/browse-by-metadata-page.component.ts (hunk 1 above)
Comment (mspalti) [reply src/app/browse-by/browse-by-metadata-page/browse-by-metadata-page.component.ts:155]: Ah, I didn't see #2128. That makes sense. Your suggestion will work. I'll make the change.

**Comment by mspalti** [comment]
@tdonohue, setting `value = ''` as the fallback does indeed fix the bug. I've made the addition.

**Review by tdonohue — APPROVED** [review]
👍 Thanks @mspalti ! This looks good. Also looks to fix the small bug introduced by https://github.com/DSpace/dspace-angular/pull/2139.  Merging immediately.

**Pull request merged**